docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,519 @@
+
+|
+
+Download Llama2 7B
+------------------
+
+In this tutorial, we'll use the HF-Format for the Llama2 7B model. For more information on checkpoint
+formats and how these are handled in TorchTune, take a look at this tutorial on checkpointing.
+
+To download the HF format Llama2 7B model, we'll use the tune CLI.
+
+.. code-block:: bash
+
+  tune download \
+  meta-llama/Llama-2-7b \
+  --output-dir <checkpoint_dir> \
+  --hf-token <ACCESS TOKEN>
+
+Make a note of ``<checkpoint_dir>``, we'll use this many times in this tutorial.
+
+|
+
+Finetune the model using LoRA
+-----------------------------
+
+For this tutorial, we'll fine-tune the model using LoRA. LoRA is a parameter efficient fine-tuning
+technique which is especially helpful when you don't have a lot of GPU memory to play with. LoRA
+freezes the base LLM and adds a very small percentage of learnable parameters. This helps keep
+memory associated with gradients and optimizer state low. Using TorchTune, you should be able to
+fine-tune a Llama2 7B model with LoRA in less than 16GB of GPU memory using bfloat16 on a
+RTX 3090/4090. For more information on how to use LoRA, take a look at our
+:ref:`LoRA Tutorial <lora_finetune_label>`.
+
+We'll fine-tune using our
+`single device LoRA recipe <https://github.com/pytorch/torchtune/blob/main/recipes/lora_finetune_single_device.py>`_
+and use the standard settings from the
+`default config <https://github.com/pytorch/torchtune/blob/main/recipes/configs/llama2/7B_lora_single_device.yaml>`_.
+
+This will fine-tune our model on the
+`Alpaca Dataset <https://github.com/pytorch/torchtune/blob/main/torchtune/datasets/_alpaca.py>`_
+using a ``batch_size=2`` and ``dtype=bfloat16``. With these settings the model
+should have a peak memory usage of ~16GB and total training time of around two hours for each epoch.
+We'll need to make some changes to the config to make sure our recipe can access the
+right checkpoints.
+
+Let's look for the right config for this use case by using the tune CLI.
+
+.. code-block:: bash
+
+    tune ls
+
+    RECIPE                                   CONFIG
+    full_finetune_single_device              llama2/7B_full_single_device
+                                             llama2/7B_full_single_device_low_memory
+                                             mistral/7B_full
+    full_finetune_distributed                llama2/7B_full
+                                             llama2/13B_full
+                                             mistral/7B_full
+    lora_finetune_single_device              llama2/7B_lora_single_device
+                                             llama2/7B_qlora_single_device
+                                             mistral/7B_lora
+    ...
+
+
+For this tutorial we'll use the ``llama2/7B_lora_single_device`` config.
+
+The config already points to the HF Checkpointer and the right checkpoint files.
```

Comment:
Even this part I feel like you can leave out and just say that we just need to point to the right tokenizer path and checkpoint directory. Separately, you could just add a note here like "to see the full config for LoRA training, you can use tune cp and view the file locally" or something like that

Reply:
Any specific reason for that? It seems like a good idea to show one example of an override?